packages/grid/_modules_/grid/components/column-header-item.tsx

```diff
@@ -44,7 +42,21 @@ export const ColumnHeaderItem = React.memo(
       });
     }
 
-    const handleResize = onResizeColumn ? () => onResizeColumn(column) : undefined;
+    const handleResize =
+      onResizeColumn && !disableColumnReorder ? () => onResizeColumn(column) : undefined;
+    const handleDragStart =
+      onColumnDragStart && !disableColumnReorder
+        ? (event) => onColumnDragStart(column, event.currentTarget)
+        : undefined;
+    const handleDragEnter = onColumnDragEnter ? (event) => onColumnDragEnter(event) : undefined;
+    const handleDragOver = onColumnDragOver
```

Comment:
same on all those functions

Reply:
I don't think that we should care about memo until we have a benchmark that shows that this is the origin of the issue. In this case, the props are applied to a primitive element so I don't see what we would gain if not extra complexity in the logic.

Reply:
Same as the previous one, the `!disableColumnReorder` check is not needed.